index.js

```diff
@@ -41,6 +42,7 @@ function Storage (chunkLength, opts) {
           file.offset = prevFile.offset + prevFile.length
         }
       }
+      if (self.path) file.path = path.resolve(path.join(self.path, self.name, file.path))
```

Comment:
This logic of resolving file paths relative to `self.path/self.name` feels too magical to me.

I think `self.path` should be the real file base path; if the user wants to put the name into the path they can do that outside.

Reply:
define too magical?

general idea here, is that the user can still specify his own full paths to files, but can also specify one that's relative to the store's path, if I added a name to the path inside webtorrent we'd get another double standard, since it wouldn't be exactly in like with what you guys did in fs-access-chunk-store